src/v/raft/state_machine_manager.h

```diff
@@ -158,7 +170,7 @@ class state_machine_manager_builder {
     template<ManagableStateMachine T, typename... Args>
     ss::shared_ptr<T> create_stm(Args&&... args) {
         auto machine = ss::make_shared<T>(std::forward<Args>(args)...);
-        _stms.push_back(machine);
+        _stms.emplace_back(T::name, machine);
```

Comment:
Do we need to squash commits c0c742d1b4f50682f561b988b2219113aec60eab and 06d95955eceae3d3299e9e6023aef43cef1dbc91 to get it to compile? Or is c0c742d1b4f50682f561b988b2219113aec60eab just ensuring we don't convert to ss::sstring for lookup?

Reply:
i was supposed to squash the two, hence the weird name